client/src/ratings_reviews/ReviewsList.jsx

```diff
@@ -10,115 +12,149 @@ class ReviewsList extends React.Component {
+      openModal: false,
+    });
   }
 
   render() {
     return (
       <div id="reviews-list">
         <div className="single-review-tile">
           <div id="sorting-option-header">
-            <h3>{helperFunction.reviewListLength(this.state.reviewsList, this.state.sortOption)}
-            <select value={this.state.sortOption} onChange={this.handleChange} id="sort-option-button">
-              <option value={"none"}>None</option>
-              <option value="relevant">Relevant</option>
-              <option value="newest">Newest</option>
-              <option value="helpful">Helpful</option>
-            </select>
+            <h3>
+              {helperFunction.reviewListLength(
+                this.state.reviewsList,
+                this.state.sortOption
+              )}
+              <select
+                value={this.state.sortOption}
+                onChange={this.handleChange}
+                id="sort-option-button"
+              >
+                <option value={"none"}>None</option>
+                <option value="relevant">Relevant</option>
+                <option value="newest">Newest</option>
+                <option value="helpful">Helpful</option>
+              </select>
             </h3>
           </div>
           {this.state.reviewsList.map((singleReview) => {
             return (
-              <ReviewTile key={singleReview.review_id} productId={this.props.currentProductId} singleReview={singleReview} />
+              <ReviewTile
+                key={singleReview.review_id}
+                productId={this.props.currentProductId}
+                singleReview={singleReview}
+              />
             );
           })}
           <div id="ratings-button-container">
             <div id="ratings-more-reviews">
-              <form onClick={this.handleClick}>
-                <button type="submit" id="more-reviews-button">MORE REVIEWS</button>
+              <form onClick={this.handleMoreReviewsClick}>
+                <button type="submit" id="more-reviews-button">
+                  MORE REVIEWS
+                </button>
               </form>
             </div>
             <div id="ratings-new-review">
-              <form>
-                <button type="submit" id="new-review-button">ADD A REVIEW +</button>
+              <form onClick={this.openNewReviewClick}>
+                <button type="submit" id="add-review-button">
+                  ADD A REVIEW +
+                </button>
               </form>
             </div>
           </div>
+          <div id="ratings-new-review-modal">
+            {this.state.openModal && (
+              <div
+                onClick={this.closeNewReviewClick}
+                className="modal-background">
+                  Hello World!
+                <NewReview
+                  productId={this.props.currentProductId}
+                  openModal={this.state.openModal}
+                  closeNewReview={this.closeNewReviewClick}
```

Comment:
is this an event handler? will it work if not using onClick?

Reply:
Great question - This functionality works because I have the onClick running on line 143, and I'm passing the closeNewReviewClick down to my NewReview component! So the event handler works!